### PR TITLE
Add test validation to import k3s cluster to rancher

### PR DESCRIPTION
**Pull request by ShylajaDevadiga**
Committer: ShylajaDevadiga<shylaja@rancher.com>

### CODE REVIEW

**Comment by ShylajaDevadiga** [comment]
@sangeethah Could you please review the PR?

File: tests/validation/tests/v3_api/common.py [hunk 1]
```diff
@@ -418,6 +418,10 @@ def get_schedulable_nodes(cluster):
     for node in nodes:
         if node.worker:
             schedulable_nodes.append(node)
+        if 'k3s' in cluster.version["gitVersion"]:
+            # Including master in list of nodes as master is also  schedulable
+            if node.controlPlane:
+                schedulable_nodes.append(node)
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/common.py]: Nitpick - Can we have this changed to accommodate in 1 if statement ?

File: tests/validation/tests/v3_api/test_import_k3s_cluster.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
+import os
+from lib.aws import AmazonWebServices
+from .common import *  # NOQA
+RANCHER_CLEANUP_CLUSTER = os.environ.get('RANCHER_CLEANUP_CLUSTER', "False")
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_import_k3s_cluster.py]: Can be inherited from common.py.

File: tests/validation/tests/v3_api/test_import_k3s_cluster.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+DATA_SUBDIR = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                           'resource')
+AWS_SSH_KEY_NAME = os.environ.get("AWS_SSH_KEY_NAME")
+K3S_VERSION = os.environ.get("K3S_VERSION", "")
+AWS_NO_OF_WORKER_NODES = os.environ.get("AWS_NO_OF_WORKER_NODES", 3)
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_import_k3s_cluster.py]: Rename to be specific to k3s

Comment (sangeethah) [reply tests/validation/tests/v3_api/test_import_k3s_cluster.py]: Make k3s specific variables start with "RANCHER"

File: tests/validation/tests/v3_api/test_import_k3s_cluster.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    cluster = validate_cluster(client, clusters[0],
+                               check_intermediate_state=False)
+    cluster_cleanup(client, cluster, aws_nodes)
+
+
+def create_nodes():
+    aws_nodes = \
+        AmazonWebServices().create_multiple_nodes(
+            AWS_NO_OF_WORKER_NODES, random_test_name("testk3s"), \
+            os_version=AWS_OS_VERSION, wait_for_ready=True)
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_import_k3s_cluster.py]: Can remove os_version and wait_for_ready parameter

File: tests/validation/tests/v3_api/test_import_k3s_cluster.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+def install_k3s_master_node(master):
+    # Connect to the node
+    ssh = ssh_call(master)
+
+    # Install master node
+    cmd = "curl -sfL https://get.k3s.io | \
+     {} sh -s - server --node-external-ip {}".\
+        format("INSTALL_K3S_VERSION={}".format(K3S_VERSION) if K3S_VERSION else "", master.public_ip_address)
+
+    stdin, stdout, stderr = ssh.exec_command(cmd, timeout=5)
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_import_k3s_cluster.py]: Can be replaced with master.execute_command(cmd)

**Pull request merged**